wnfs/src/private/privateref.rs

```diff
@@ -156,30 +167,67 @@ impl Debug for PrivateRef {
 
         f.debug_struct("PrivateRef")
             .field("saturated_name_hash", &sat_name_hash_str)
-            .field("content_key", &self.content_key.0)
-            .field("revision_key", &self.revision_key.0)
+            .field("temporal_key", &self.temporal_key.0)
+            .field("content_cid", &self.content_cid)
             .finish()
     }
 }
 
+impl RevisionRef {
+    /// Creates a RevisionRef from provided namefilter, ratchet seed and inumber.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use wnfs::{private::RevisionRef, Namefilter};
+    /// use rand::{thread_rng, Rng};
+    ///
+    /// let rng = &mut thread_rng();
+    /// let revision_ref = RevisionRef::with_seed(
+    ///     Namefilter::default(),
+    ///     rng.gen::<[u8; 32]>(),
+    ///     rng.gen::<[u8; 32]>(),
+    /// );
+    ///
+    /// println!("Private ref: {:?}", revision_ref);
+    /// ```
+    pub fn with_seed(name: Namefilter, ratchet_seed: HashOutput, inumber: HashOutput) -> Self {
+        PrivateNodeHeader::with_seed(name, ratchet_seed, inumber).derive_revision_ref()
+    }
+
+    /// Turns a reivison ref into a more specific pointer, a private ref.
+    ///
+    /// The revision ref refers to a whole multivalue that may or may not exist
+    /// or may refer to multiple private nodes.
+    ///
+    /// The resulting private ref refers to the given CID in the multivalue.
+    pub fn as_private_ref(self, content_cid: Cid) -> PrivateRef {
+        PrivateRef {
+            saturated_name_hash: self.saturated_name_hash,
+            temporal_key: self.temporal_key,
+            content_cid,
+        }
+    }
+}
+
 //--------------------------------------------------------------------------------------------------
 // Tests
 //--------------------------------------------------------------------------------------------------
 
 #[cfg(test)]
 mod tests {
     use crate::{
-        private::PrivateForest,
         utils::{self, test_setup},
         PrivateDirectory, PrivateNode,
     };
     use chrono::Utc;
+    use futures::StreamExt;
 
-    use super::PrivateRef;
+    use super::RevisionRef;
 
     #[async_std::test]
-    async fn can_create_privateref_deterministically_with_user_provided_seeds() {
-        let (forest, store, rng) = test_setup::init!(forest, mut store, mut rng);
+    async fn can_create_revisionref_deterministically_with_user_provided_seeds() {
+        let (ref mut forest, store, rng) = test_setup::init!(forest, mut store, mut rng);
```

Comment:
Nit: I would like us to stick to a convention here. So we could use `ref mut` everywhere or we could support `mut forest` in the macro impl.